homeassistant/components/elgato/light.py

```diff
@@ -143,12 +135,12 @@ async def async_update(self) -> None:
             self._state = None
 
     @property
-    def device_info(self) -> dict[str, Any]:
+    def device_info(self) -> DeviceInfo:
         """Return device information about this Elgato Key Light."""
         return {
-            ATTR_IDENTIFIERS: {(DOMAIN, self._info.serial_number)},
-            ATTR_NAME: self._info.product_name,
-            ATTR_MANUFACTURER: "Elgato",
-            ATTR_MODEL: self._info.product_name,
-            ATTR_SOFTWARE_VERSION: f"{self._info.firmware_version} ({self._info.firmware_build_number})",
+            "identifiers": {(DOMAIN, self._info.serial_number)},
+            "name": self._info.product_name,
+            "manufacturer": "Elgato",
+            "model": self._info.product_name,
+            "sw_version": f"{self._info.firmware_version} ({self._info.firmware_build_number})",
```

Comment:
I don't see why this change is needed. Is this a MyPy bug? There is nothing in PEP-0589 that tells me that this is needed?

Reply:
From mypy docs:
> Mypy will also reject a runtime-computed expression as a key, as it can’t verify that it’s a valid key. You can only use string literals as TypedDict keys.

Reply:
That's for the TypedDict though. Not sure if that applies to the thing that is checked.

Reply:
I think they mean that only string literals works in the dicts that are checked.

Reply:
But that renders any form of dynamically generated dictionaries with any typeddict completely useless?